src/cephadm/cephadmlib/daemons/smb.py

```diff
@@ -104,20 +106,29 @@ def conf_interface(self) -> str:
 
 
 class Ports(enum.Enum):
-    SMB = 445
-    SMBMETRICS = 9922
-    CTDB = 4379
-    REMOTE_CONTROL = 54445
+    SMB = ceph.smb.constants.SMB_PORT
+    SMBMETRICS = ceph.smb.constants.SMBMETRICS_PORT
+    CTDB = ceph.smb.constants.CTDB_PORT
+    REMOTE_CONTROL = ceph.smb.constants.REMOTE_CONTROL_PORT
 
     def customized(self, service_ports: Dict[str, int]) -> int:
         """Return a custom port value if it is present in service_ports or the
         default port value if it is not present.
         """
-        port = service_ports.get(self.name.lower())
+        port = service_ports.get(str(self))
         if port:
             return int(port)
         return int(self.value)
 
+    def __str__(self) -> str:
+        names = {
+            self.SMB: ceph.smb.constants.SMB,
+            self.SMBMETRICS: ceph.smb.constants.SMBMETRICS,
+            self.CTDB: ceph.smb.constants.CTDB,
+            self.REMOTE_CONTROL: ceph.smb.constants.REMOTE_CONTROL
+        }
+        return names[self]
```

Comment:
mypy complaint from make check (`check-black` was also failing)
```
cephadmlib/daemons/smb.py: note: In member "__str__" of class "Ports":
cephadmlib/daemons/smb.py:130: error: Invalid index type "Ports" for "dict[int, str]"; expected type "int"  [index]
Found 1 error in 1 file (checked 49 source files)
mypy: exit 1 (13.13 seconds) /home/jenkins-build/build/workspace/ceph-pull-requests/src/cephadm> mypy --config-file ../mypy.ini cephadm.py cephadmlib pid=775889
mypy: FAIL ✖ in 33.01 seconds
```

Reply:
This was a weird one (see coment). But I worked around it.